src/Models/ShallowWaterModels/shallow_water_model.jl

```diff
@@ -112,7 +112,7 @@ Keyword arguments
 function ShallowWaterModel(;
                            grid,
                            gravitational_acceleration,
-                               clock = Clock{eltype(grid)}(0, 0, 1),
+                               clock = Clock{eltype(grid), eltype(grid)}(0, Inf, 0, 1),
```

Comment:
Personally I would add a constructor for `Clock` that takes one type rather than writing this. But you can decide.

Reply:
Also note that we might reasonably expect users to provide custom clocks with their own time-step and iteration --- but that the stage and time-step probably won't commonly be provided when building a clock. So I think it would make more sense to have a syntax that looks like

```julia
Clock{TT}(time, iteration, [optional args])
```

Reply:
I think the type of the time-step can probably be inferred from the time type `TT` (ie if the time type is either a floating point or a DateTime or something like that)

Reply:
There actually is already a constructor I can just change it a bit to infer the `DT` https://github.com/CliMA/Oceananigans.jl/blob/3bb62a647a55a7dadf5f37331321bf0020a78c4d/src/TimeSteppers/clock.jl#L27

Reply:
It seems like `Dates` only stores things as integers and is never directly used but the timestep is always `eltype(grid)` so I'm not sure it can be directly infered.

Reply:
We don't support dates right now -- to support it, we have to generalize how the time-step specification works.

Reply:
> the timestep is always eltype(grid) so I'm not sure it can be directly infered.

We will have to change how time-step specification works to support DateTime properly.